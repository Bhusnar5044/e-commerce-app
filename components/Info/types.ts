import { Product } from '@/types';

export type IProps = {
    data: Product;
};
