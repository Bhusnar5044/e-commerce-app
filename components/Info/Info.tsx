'use client';

import { ShoppingCart } from 'lucide-react';

import Button from '@/components/ui/Button';
import Currency from '@/components/ui/Currency';
import { addItem } from '@/redux/features/cartSlice';
import { useAppDispatch } from '@/redux/store';
import { memo } from 'react';
import { IProps } from './types';

export const Info: React.FC<IProps> = memo(({ data }) => {
    const dispatch = useAppDispatch();

    const onAddToCart = () => {
        dispatch(addItem(data));
    };

    return (
        <div>
            <h1 className="text-3xl font-bold text-gray-900">{data.title}</h1>
            <h3 className="font-semibold text-black">{data.description}</h3>
            <div className="mt-3 flex items-end justify-between">
                <p className="text-2xl text-gray-900">
                    <Currency value={data?.price} />
                </p>
            </div>
            <hr className="my-4" />
            <div className="mt-10 flex items-center gap-x-3">
                <Button onClick={onAddToCart} className="flex items-center gap-x-2">
                    Add To Cart
                    <ShoppingCart size={20} />
                </Button>
            </div>
        </div>
    );
});

Info.displayName = 'Info';
