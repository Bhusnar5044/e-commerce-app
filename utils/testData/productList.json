[
    {
        "id": "1",
        "category": { "id": "1", "name": "Glasses" },
        "price": "12",
        "isFeatured": true,
        "name": "Dummy",
        "size": { "id": "1", "name": "S", "value": "12" },
        "color": { "id": "1", "name": "Gray", "value": "gray" },
        "images": [{ "id": "1", "url": "" }]
    }
]
