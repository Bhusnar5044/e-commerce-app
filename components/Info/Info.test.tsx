import { renderWithStore } from '@test-utils';
import * as ProductsMock from '@utils/testData/productList.json';
import { Info } from './Info';

describe('Info', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<Info data={ProductsMock[0]} />);
        expect(container).toMatchSnapshot();
    });
});
