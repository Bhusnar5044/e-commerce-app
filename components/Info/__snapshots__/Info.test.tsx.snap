// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should render properly 1`] = `
<div>
  <div>
    <h1
      class="text-3xl font-bold text-gray-900"
    >
      iPhone 9
    </h1>
    <h3
      class="font-semibold text-black"
    >
      An apple mobile which is nothing like apple
    </h3>
    <div
      class="mt-3 flex items-end justify-between"
    >
      <p
        class="text-2xl text-gray-900"
      >
        <div
          class="font-semibold"
        >
          $549.00
        </div>
      </p>
    </div>
    <hr
      class="my-4"
    />
    <div
      class="mt-10 flex items-center gap-x-3"
    >
      <button
        class="w-auto rounded-full bg-black border border-transparent px-5 py-3 disabled:cursor-not-allowed disabled:opacity-50 text-white font-semibold hover:opacity-75 transition flex items-center gap-x-2"
        type="button"
      >
        Add To Cart
        <svg
          class="lucide lucide-shopping-cart"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="21"
            r="1"
          />
          <circle
            cx="19"
            cy="21"
            r="1"
          />
          <path
            d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
